src/lib/xmlparse.rs

```diff
@@ -338,7 +338,7 @@ trait XmlHandlers {
     unsafe fn skippedEntity(&self, _: *const XML_Char, _: c_int) -> bool;
     unsafe fn startCDataSection(&self) -> bool;
     unsafe fn startDoctypeDecl(&self, a: *const XML_Char, b: *const XML_Char, c: *const XML_Char, d: c_int) -> bool;
-    unsafe fn startElement(&self, _: *const XML_Char, _: *mut ATTRIBUTE) -> bool;
+    unsafe fn startElement(&self, _: *const XML_Char, _: &mut [[*const XML_Char; 2]]) -> bool;
```

Comment:
Should this be mutable? I don't think anything in expat assumes that the callback will modify the array.

Reply:
The original C signature is
```C
typedef void(XMLCALL *XML_StartElementHandler)(void *userData,
                                               const XML_Char *name,
                                               const XML_Char **atts);
```
so in theory C code could modify the pointers. I think it's risky (aka UB) to have the pointers be constant on one side of the FFI and mutable on the other.

Reply:
Alright, let's keep this mutable. We'll probably need to audit the FFI for stuff like this at some point and make sure pointers have sufficient mutability (or don't matter).